client/src/includes/breadcrumbs.test.js

```diff
@@ -0,0 +1,33 @@
+import initCollapsibleBreadcrumbs from './breadcrumbs';
+
+describe('initCollapsibleBreadcrumbs', () => {
+  jest.spyOn(document, 'addEventListener').mockImplementation(() => {});
+  it('should do nothing if there is no breadcrumbs container', () => {
+    // Set up our document body
+    document.body.innerHTML =
+      '<div>' +
+      '  <span id="username" />' +
+      '  <button id="button" />' +
+      '</div>';
+    initCollapsibleBreadcrumbs();
+    // no event listeners registered
+    expect(document.addEventListener).not.toHaveBeenCalled();
+  });
+
+  it('should expand the breadcrumbs when clicked', () => {
```

Comment:
One thing you can do is wrap these two tests in a nested describe block. 

At the stat of the describe block you at up your DOM. 

Then run test a. That the event listeners get added. 
Test b - that the click expands. 
Test C - that the click again collapses.

Reply:
I'll do that

Reply:
Is there a way I can access variables defined in breadcrumbs.js from tests? 
I'm talking about these variables defined here https://github.com/wagtail/wagtail/blob/b53f12d3944c699ff76a39ea00cb5b1dec7b4f2d/client/src/includes/breadcrumbs.js#L19


I want to test the value of `keepOpen` in my test

Reply:
Not really - and that's a good thing. 

You want to test the behaviour (changing of attributes or classes) based on interaction (click / mouse events). 

You don't want to test the implementation (internal state) as this makes your tests too brittle if these change.

Reply:
Got it